version: '3.4'
services:
  web:
    tty: true
    shm_size: 1gb
    privileged: true
    image: my_odoo_14:latest
    depends_on:
      - db
      - wdb
    ports:
      - "8069:8069"
      - "8072:8072"
      # - "5678:5678"
      # - "1984:1984"
    volumes:
      - odoo-web-data:/var/lib/odoo
      - ./config:/etc/odoo
      - /home/admin/git_reps/fsa-parser:/mnt/1
      # - ../../../odoo/odoo-14.0/odoo/odoo/fields.py:/usr/lib/python3/dist-packages/odoo/fields.py
      # - ../../../odoo/odoo-14.0/odoo/odoo/models.py:/usr/lib/python3/dist-packages/odoo/models.py
      # - ../../../odoo/odoo-14.0/odoo/odoo/api.py:/usr/lib/python3/dist-packages/odoo/api.py
    environment:
      - WDB_SOCKET_SERVER=wdb
      - WDB_NO_BROWSER_AUTO_OPEN=True
    #    command: ['-d', '--log-handler=odoo.addons.queue_job:DEBUG', '--workers=3']
    # - command: 
    # command: python -m debugpy --listen odoo:5678 --log-to-stderr ./odoo-bin -c /config/odoo.conf --dev=all,xml
    # command: bash -c "/entrypoint.sh odoo --dev=xml" 
    command: bash -c "/entrypoint.sh odoo"
    # command: python3 -m debugpy --listen web:5678 --log-to-stderr /usr/bin/odoo -c /etc/odoo/odoo.conf --db_host=db --db_port=5432 --db_user=odoo --db_password=odoo --dev=all,xml
    # command: python3 -m debugpy --listen web:5678 --log-to-stderr /usr/bin/odoo -c /etc/odoo/odoo.conf --db_host=db --db_port=5432 --db_user=odoo --db_password=odoo --dev=all,xml

  db:
    image: postgres:10
    ports:
      - "5432:5432"
    environment:
      - POSTGRES_DB=postgres
      - POSTGRES_PASSWORD=peacedataodoo
      - POSTGRES_USER=odoo
      - PGDATA=/var/lib/postgresql/data/pgdata14
    volumes:
      - odoo-db-data14:/var/lib/postgresql/data/pgdata14
  wdb:
    image: kozea/wdb:3.3.0
    ports: ['1984:1984']

volumes:
  odoo-web-data:
  odoo-db-data14:
